modules/common/endpoint/endpoint.go

```diff
@@ -131,6 +133,8 @@ func ExposeEndpoints(
 				exportLabels,
 				timeout,
 			)
+			// add annotation to register service name in dnsmasq
+			svc.AddAnnotation(map[string]string{AnnotationHostnameKey: svc.GetServiceHostname()})
```

Comment:
Is there a technical reason why we add this annotation after NewService call while the rest of the annotations collected before the NewService call? If no specific reason then I would move this closer to the rest of the annotation gathering. Also the new AddAnnotation function could be used for the other annotations above too.

Reply:
its because the NewService initializes the serviceHostname. We rely on that information here as the value of the annotation.  On the 2nd point you are right, we first initialize the svc and then use AddAnnotation to add the  MetalLB annotations.